syntax = "proto3";

package aea.bosch.perun_grpc.v0_1_0;

message PerunGrpcMessage{

  // Performatives and contents
  message Request_Performative{
    string type = 1;
    bytes content = 2;
  }

  message Response_Performative{
    string type = 1;
    bytes content = 2;
  }

  message Paychresp_Performative{
    string session_id = 1;
    string type = 2;
    bytes content = 3;
  }

  message Accept_Performative{
  }

  message Reject_Performative{
  }

  message End_Performative{
  }


  oneof performative{
    Accept_Performative accept = 5;
    End_Performative end = 6;
    Paychresp_Performative paychresp = 7;
    Reject_Performative reject = 8;
    Request_Performative request = 9;
    Response_Performative response = 10;
  }
}
